java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageSerializer.java

```diff
@@ -380,6 +403,16 @@ public static ArrowBlock serialize(WriteChannel out, ArrowDictionaryBatch batch)
     return new ArrowBlock(start, metadataLength + 4, bufferLength);
   }
 
+  /** Returns the serialized flatbuffer bytes for this object. */
+  public static ByteBuffer serializeMetadata(ArrowDictionaryBatch batch) {
```

Comment:
could you make one generic `serializeMetadata(ArrowMessage message, byte headerType)` instead of 3 different ones?

Reply:
Schema doesn't implement ArrowMessage and I'd prefer not to make the change necessary to do it.  I have consolidated the two batch types by adding a getType to ArrowMessage interface.

Reply:
ok, sounds fine